conda/plugins/specs.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
```

Comment:
I like the idea of putting the specs and plugin objects into a separate file, I think this hadn't been done because it wasn't clear how we'll evolve the code. I wonder how quickly this would become unwieldy, the more plugin hooks we support, though. Maybe it would be best to separate them by use case, e.g. put the subcommand specific code into `conda.plugins.subcommands`? Will the use of the @_hookspec decorator enough to trigger the registration of the plugin hook spec then?

Reply:
I only did this because `pluggy.PluginManager.add_hookspecs` expects a namespace, either a module or a class. AFAIK we need both the `@_hookspec` and the `pluggy.PluginManager.add_hookspecs` for the specs to be registered.

I wasn't super happy with using `conda.plugins.specs`, not sure about the directory structure, perhaps we should remove this specs submodule until we have more types of plugins and a better feel for how we want this laid out?